src/main/java/com/taskadapter/redmineapi/internal/comm/betterssl/BetterSSLFactory.java

```diff
@@ -52,6 +53,30 @@ public static SSLSocketFactory createSocketFactory(Collection<KeyStore> extraSto
 		}
 	}
 
+        public static SSLSocketFactory createSocketFactory(KeyStore keystore, String keystorePassword, Collection<KeyStore> extraStores) throws KeyStoreException, KeyManagementException {
+		final Collection<X509TrustManager> managers = new ArrayList<>();
+
+                for (KeyStore ks : extraStores) {
+			addX509Managers(managers, ks);
+		}
+		/* Add default manager. */
+		addX509Managers(managers, null);
+		final TrustManager tm = new CompositeTrustManager(managers);
+
+		try {
+                    final KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                    kmf.init(keystore, keystorePassword.toCharArray());
+                    final SSLContext ctx = SSLContext.getInstance("TLSv1.2");
```

Comment:
why this algorithm specifically?

Reply:
Basically anything below TLSv1.2 is deprecated, but let me check if plain "SSL" will upgrade connection status to TLSv1.2/1.3 if supported. Thanks!

Reply:
Ok, according to https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#SSLContext , we could use "TLS" to let it negotiate against any supported TLS version in the JVM. There's no documentation regarding a catch-all, so we've got 2 options: Parametrize it, or set it to 'TLS'. What would you prefer?

Reply:
TLS should be fine for now. no one requested this before

Reply:
Done, changed to TLS.